src/components/selection_input/selection_input.ts

```diff
@@ -224,6 +224,11 @@ export class SelectionInput extends Component<Props, SpreadsheetChildEnv> {
 
   removeInput(rangeId: number) {
     const index = this.store.selectionInputs.findIndex((range) => range.id === rangeId);
+    // check first if the range is empty & remove it directly from the store because it's not yet confirmed
+    if (!this.store.selectionInputs[index].xc) {
+      this.store.removeRange(rangeId);
+      return;
+    }
     this.props.onSelectionRemoved?.(index);
     this.props.onSelectionConfirmed?.();
```

Comment:
Kinda strange to have a different behaviour is the range is empty or not. Why does removing a non-empty range confirms the selection input, but removing a non empty-range does not ? The condition should probably be on whether or not the selection input is focused.

Reply:
Probably linked to this: in the screenshot, the "delete" button does not work on the "A1" range

![image](https://github.com/user-attachments/assets/b014aeba-62c2-498e-8498-0a7914ad5021)